AEPCore/Sources/rules/LaunchRulesEngine.swift

```diff
@@ -104,6 +107,7 @@ public class LaunchRulesEngine {
     }
 
     private func evaluateRules(for event: Event) -> Event {
+        let dispatchChainCount = dispatchChainedEventsCount.removeValue(forKey: event.id)
```

Comment:
The id should not be removed until it is evaluated and the chain needs to be updated.

Reply:
I remove the id here so the list gets cleaned up in cases where the event does not trigger the dispatch consequence event. In cases where no rules matched the event or the event triggers an add or modify data consequence, the id should get removed as the event was evaluated.  It is my understanding that events are evaluated for rules once, so once the `evaluateRules` function completes, the event id no longer needs to be tracked.

Reply:
This seems fair except for the case where an event is created and then dispatched multiple times, the Event UUID does not change in this case.

Reply:
In what cases would an event get dispatched multiple times? It's possible for anyone to dispatch a received event, but I'm not familiar with where in the SDK that occurs.

Reply:
💭  This scenario may cause some other interesting behavior in the EventHub, if multiple events are queued consecutively the last one will override the event number of the previous one before.

Reply:
Looks like this is not prevented today, so you could have something like:
```
/// some pre-computed event
let event: Event = Event(name: "test", type: "test.type", source: "test.source", data: ["some":"data"])
...
func click() {
    MobileCore.dispatch(event: event)
}
```

Reply:
Remember this list only contains the UUID of events dispatched from this Rules Engine, so we'd need someplace else to be re-dispatching these events to the Event Hub for them to be evaluated multiple times. In your scenario, events being evaluated will not override the event UUIDs in the list, but it will cause the count set to new events created from the dispatch consequence to be set to 0.

Reply:
A compromise would be to keep the triggering event UUID in the list if the event matches a dispatch consequence. If the the event does not match a dispatch consequence, it is removed from the list. This will mean events which match a dispatch consequence rule will always stay in the list. As this is an unexpected case, it is assumed it will be a small number of events.
One question here is if the event is evaluated a second time, should the dispatch consequence get processed again? On the one hand, the consequence was already processed when the event as first received. On the other hand, we'd probably want the SDK to behave the same if the same event is dispatched multiple times, which would include processing the same rules the event matches.

Reply:
I feel calling `MobileCore.dispatch(event: event)` should behave the same way and these checks were to limit the events automatically dispatched by rules. We should create a separate task and look into handling dispatching same event multiple times.

Reply:
@praveek added https://github.com/adobe/aepsdk-core-ios/issues/688 for this.